### PR TITLE
Support all bitwise query operators

### LINKED ISSUE #256: Support all bitwise query operators
Support all query comparison operators for all data types.

- [x] $bitsAllClear
- [x] $bitsAllSet
- [x] $bitsAnyClear
- [x] $bitsAnySet

### PR DESCRIPTION
**Pull request by w84thesun**
Closes #256.

### CODE REVIEW

**Comment by w84thesun** [comment]
In order to tests pass I need this PR to be merged → https://github.com/AlekSi/test_db/pull/12

**Review by AlekSi — COMMENTED** [review]
Tests fail

File: internal/handlers/common/error.go [hunk 1]
```diff
@@ -119,3 +119,25 @@ func ProtocolError(err error) (*Error, bool) {
...
 )
+
+// formatBitwiseOperatorErr formats protocol error for given internal error and bitwise operator.
+// Mask value used in error message.
+func formatBitwiseOperatorErr(err error, operator string, maskValue any) error {
+	switch err {
+	case errNotWholeNumber:
+		return NewErrorMsg(ErrFailedToParse,
+			fmt.Sprintf("Expected an integer: %s: %#v", operator, maskValue),
+		)
```
Comment (AlekSi) [inline internal/handlers/common/error.go:130]:
```suggestion
		return NewErrorMsg(
		    ErrFailedToParse,
			fmt.Sprintf("Expected an integer: %s: %#v", operator, maskValue),
		)
```

For the same reason we don't write

```go
func foo() { codeLine1
  codeLine2
}
```

The same in other places

File: internal/handlers/common/error.go [hunk 2]
```diff
@@ -119,3 +119,25 @@ func ProtocolError(err error) (*Error, bool) {
 var (
```
Comment (AlekSi) [inline internal/handlers/common/error.go:119]: Interface checking should be at the very end of the file

File: internal/handlers/common/params.go [hunk 3]
```diff
@@ -132,9 +130,7 @@ func getBinaryMaskParam(value any) (types.Binary, error) {
 	case float64:
 		// TODO check float negative zero
 		if value != math.Trunc(value) || math.IsNaN(value) || math.IsInf(value, 0) {
-			return types.Binary{}, NewErrorMsg(ErrFailedToParse,
-				fmt.Sprintf("Expected an integer: $bitsAllClear: %#v", value),
-			)
+			return types.Binary{}, errNotWholeNumber
```
Comment (AlekSi) [inline internal/handlers/common/params.go:133]: The current value of `errNotWholeNumber` starts with a wrong function name. I think we could remove those prefixes if they are not correct

File: internal/handlers/handler_test.go [hunk 4]
```diff
@@ -1128,6 +1128,153 @@ func TestFind(t *testing.T) {
...
+		"BitsAllSet": {
+			schemas: []string{"values"},
+			req: must.NotFail(types.NewDocument(
+				"find", "values",
+				"filter", must.NotFail(
+					types.NewDocument(
+						"name", "int32",
+						"value", must.NotFail(types.NewDocument(
+							"$bitsAllSet", int32(42))),
+					)),
```
Comment (AlekSi) [inline internal/handlers/handler_test.go:1140]:
```suggestion
				"filter", must.NotFail(types.NewDocument( // two opening parenthesis
						"name", "int32",
						"value", must.NotFail(types.NewDocument( // two opening parenthesis
							"$bitsAllSet", int32(42),
						)), // two closing parenthesis
					)), // two closing parenthesis
```

or

```suggestion
				"filter", must.NotFail(
				    types.NewDocument(
						"name", "int32",
						"value", must.NotFail(
							types.NewDocument(
								"$bitsAllSet", int32(42),
							),
						),
					),
				),
```

not a wild mix

File: internal/handlers/common/params.go [hunk 5]
```diff
@@ -70,6 +70,8 @@ func AssertType[T types.Type](value any) (T, error) {
 var (
 	errUnexpectedType = fmt.Errorf("GetWholeNumberParam: unexpected type")
 	errNotWholeNumber = fmt.Errorf("GetWholeNumberParam: not a whole number")
+	errNegativeNumber = fmt.Errorf("getBinaryMaskParam: negative number")
```
Comment (seeforschauer) [inline internal/handlers/common/params.go:73]: could some context be added to the error? i.e. that it's for bitwise op

**Review by seeforschauer — CHANGES_REQUESTED** [review]
tests fail

File: internal/handlers/common/params.go (hunk 5 above)
Comment (w84thesun) [reply internal/handlers/common/params.go]: It might be used not just for bitwise operators.

Comment (seeforschauer) [reply internal/handlers/common/params.go]: ok

File: internal/handlers/common/error.go (hunk 2 above)
Comment (w84thesun) [reply internal/handlers/common/error.go]: Fixed

File: internal/handlers/common/params.go (hunk 3 above)
Comment (w84thesun) [reply internal/handlers/common/params.go:133]: Fixed

**Review by seeforschauer — APPROVED** [review]

File: internal/handlers/handler_test.go (hunk 4 above)
Comment (w84thesun) [reply internal/handlers/handler_test.go]: Checked all tests, fixed to 2 opening/2 closing

**Review by AlekSi — APPROVED** [review]

**Pull request merged**